### ISSUE TITLE
Multiple consecutive icon requests for HTTP 403

### ISSUE THREAD

**Issue by michaelnordmeyer** [jocmp/capyreader#495]
### Steps to Reproduce

1. Subscribe to a feed
2. Monitor server log file for 403s
3. Initiate redrawing of feed list through, for example, rotating the phone with screen rotation on

Step 3 will request as many icons as there are feed entries visible. In portrait maybe 7, in landscape maybe 3. But on each and every rotation.

This behavior might apply to 404s as well, but I didn't test it. It stops after the icon has been successfully received.

### Expected behavior

A feed icon is only fetched on feed subscription and feed update events.

### Device, Operating system, and app version

Capy Reader 2024.11.1070

**Comment by jocmp** [comment]
hey @michaelnordmeyer thanks for the bug report. Can you share a link to the feed that 403's so I can try out the fix with it?

**Comment by michaelnordmeyer** [comment]
Unfortunately not, because it's a private feed. But you can return a 403 for a user-agent beginning with `okhttp` using a regex (see related bug #496). In nginx it would be adding a single line to the beginning of the main `location / {}` block:

```
location / {
  if ($http_user_agent ~ "^okhttp") { return 403; }
  [rest of this location including tryfiles]
}
```

A test feed could have a feed icon called `icon.png` and look like this:

```xml
<?xml version="1.0" encoding="utf-8"?>
<feed xmlns="http://www.w3.org/2005/Atom" xml:lang="en-US">
  <link href="https://example.com/feed.xml" rel="self" type="application/atom+xml" />
  <link href="https://example.com/" rel="alternate" type="text/html" hreflang="en-US" />
  <updated>2024-11-12T12:00:00+00:00</updated>
  <id>https://example.com/feed.xml</id>
  <icon>https://example.com/icon.png</icon>
  <title>Title</title>
  <subtitle>Subtitle</subtitle>
  <author>
    <name>Author</name>
    <uri>https://example.com/</uri>
  </author>
  <entry>
    <title>Post</title>
    <link href="https://example.com/post" rel="alternate" type="text/html" title="Post" />
    <published>2024-11-12T12:00:00+00:00</published>
    <updated>2024-11-12T12:00:00+00:00</updated>
    <id>https://example.com/post</id>
    <author>
      <name>Author</name>
      <uri>https://example.com/</uri>
    </author>
    <category term="Test" />
    <summary>Test summary.</summary>
    <content type="html" xml:base="https://example.com/post"><![CDATA[Test content.]]></content>
  </entry>
</feed>
```

Replace `example.com` with your URL.

**Comment by michaelnordmeyer** [comment]
My guess is, that for every action that tries to display or layout the feed list entries, feed entry resources will be requested if missing, which is why it might happen for 404s as well.

**Comment by michaelnordmeyer** [comment]
I created a feed for testing on a server, which will block the generic `okhttp/x.y.z` user-agent: https://lab.michaelnordmeyer.com/capyreader/feed.xml

This is the web server log output just after subscribing to a single-post feed:

```text
[12/Nov/2024:19:41:17.038 +0000] 200 "GET /capyreader/feed.xml HTTP/2.0" "CapyReader (RSS Reader https://capyreader.com/) okhttp/4.12.0"
[12/Nov/2024:19:41:17.389 +0000] 403 "GET /capyreader/icon.png HTTP/2.0" "okhttp/4.12.0"
[12/Nov/2024:19:41:17.390 +0000] 403 "GET /capyreader/icon.png HTTP/2.0" "okhttp/4.12.0"
```

And after a single rotation from portrait to landscape mode:

```text
[12/Nov/2024:19:43:23.662 +0000] 403 "GET /capyreader/icon.png HTTP/2.0" "okhttp/4.12.0"
[12/Nov/2024:19:43:23.664 +0000] 403 "GET /capyreader/icon.png HTTP/2.0" "okhttp/4.12.0"
```

And after scrolling the "All" view until the post is displayed:

```text
[12/Nov/2024:19:47:17.467 +0000] 403 "GET /capyreader/icon.png HTTP/2.0" "-" "okhttp/4.12.0"
```

**Comment by jocmp** [comment]
@michaelnordmeyer wow, thanks for setting that endpoint up. I'll use it when I put up a fix.

**Comment by jocmp** [comment]
I have a fix up for this as of [2024.11.1074-dev](https://github.com/jocmp/capyreader/releases/tag/2024.11.1074-dev). I also copied over the test feed with a non-existent png to test out 404's here https://jocmp.com/fake-feed.xml. If you use F-Droid or Google Play it should be live in the next week.

Please reopen this if you keep seeing the issue and thanks again for all the test setup. Super helpful.
